server/src/server.ts

```diff
@@ -135,6 +145,17 @@ function validateTextDocument(textDocument: TextDocument): void {
   }
 }
 
+connection.onDefinition((pos: TextDocumentPositionParams) => {
+  const doc: TextDocument = documents.get(pos.textDocument.uri);
+  const linePrefix = doc.getText().substr(0, doc.offsetAt(pos.position));
+  const lastToken = linePrefix.match(/\b([^:\.\s]+)\b$/g);

```

Comment:
* You might have a “whole word” flag available to avoid explicitly using `\b` on both sides.
* The `\b$` at the end seems a bit redundant, don't you think? :thinking: Well, actually, it does not mean _exactly_ the same thing than just `$`:

![image](https://user-images.githubusercontent.com/36635838/53250113-0bb8fb00-36ba-11e9-8c46-77ff89491bbb.png)

Indeed, to have a “boundary”, you need the last character to be a “word character”.
_Shikashi_, do you _want_ this distinction to apply here? Do you _need_ a word character to appear at the end of your `[^:\.\s]+`? I'd consider doing one of these things:

* turning `[^:\.\s]+` into `[^:\.\s]*\w` and getting rid of the `\b` to state explictly that you need this thing to and with a word character. But I'm not even sure you need one; I'll let you tell me.
* just getting rid of the final `\b` in favor of a lone `$`, if you do not need a word character at the end.

### By the way

Why `\.` within the character list? Most metacharacters do not need to be escaped when they appear in this context. Generally, only square brackets cause a bit of trouble.

![image](https://user-images.githubusercontent.com/36635838/53250566-f42e4200-36ba-11e9-85fd-0df9df15b186.png)


### By the way again

What is this capturing group (`( … )`) for?